src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -108,6 +108,11 @@ public static int Main(string[] args)
 
                     foreach (var rule in analyzer.SupportedDiagnostics)
                     {
+                        if (rule.Category == "Globalization")
+                        {
+                            Debug.Assert(analyzerType.BaseType?.Name == "AbstractGlobalizationDiagnosticAnalyzer", $"Analyzer {analyzerType.Name} was expected to inherit AbstractGlobalizationDiagnosticAnalyzer.");
```

Comment:
This may need to be converted to a loop (i.e. checking the inheritance hierarchy, not only the direct parent)